Realm/RLMSyncUtil.h

```diff
@@ -59,6 +65,16 @@ typedef RLM_ERROR_ENUM(NSInteger, RLMSyncError, RLMSyncErrorDomain) {
 
     /// An error that indicates an internal, unrecoverable error with the underlying synchronization engine.
     RLMSyncErrorClientInternalError     = 6,
+
+    /// An error that indicates the Realm needs to be reset.
+    /// There are two keys in the `userInfo` dictionary of particular interest.
+    /// `kRLMSyncRecoveredRealmLocationPathKey` describes the path of the recovered copy of the Realm. It will not
+    /// actually exist until the app is restarted or the recovery block is called, whichever happens first.
+    /// `kRLMSyncImmediatelyRecoverRealmBlockKey` contains a recovery block that takes no arguments and returns nothing.
+    /// If your app can guarantee that all instances of the offending Realm can be closed and invalidated, you may call
+    /// this block to immediately reset the Realm. This creates a recovery copy and deletes the original files, allowing
+    /// the Realm to be re-opened and a fresh copy to be re-downloaded.
```

Comment:
What happens if you don't immediately call the block or if you call the block some time later is unclear. A brief mention of why a Realm would need to be reset could help as well.